homework/unique_ptr/UniquePointer.hpp

```diff
@@ -0,0 +1,90 @@
+#pragma once
+
+class NullPtrException : public std::runtime_error {
+public:
+    NullPtrException(std::string msg) : std::runtime_error{msg} {}
+};
+
+template <typename T>
+class UniquePointer {
+public:
+    UniquePointer(T* ptr);
+    UniquePointer();
+    ~UniquePointer();
+    UniquePointer(const UniquePointer&) = delete;
+    UniquePointer(UniquePointer&& anotherPtr);
+
+    T* get();
+    T* release();
+    void reset(T* ptr);
+
+    T operator*() const;
```

Comment:
```suggestion
    T& operator*() const;
```
This operator should return a reference, not a copy